bodhi/server/consumers/masher.py

```diff
@@ -1289,10 +1320,18 @@ def _generate_module_list(self):
 
         # make sure that the modules we want to update get their correct versions
         for update in self.compose.updates:
-            for build in update.builds:
-                nsv = build.nvr.rsplit('-', 1)
-                ns = nsv[0]
-                version = nsv[1]
+            # We need to get the Koji builds also for the updates.
+            koji.multicall = True
+            for update in self.compose.updates:
+                for build in update.builds:
+                    koji.getBuild(build.nvr)
+            results = koji.multiCall()
+            for result, build in zip(results, update.builds):
+                self._raise_on_get_build_multicall_error(result, build)
+                koji_build = result[0]
+                # name:stream:version maps to Koji's name-version-release.
+                ns = "%s:%s" % (koji_build["name"], koji_build["version"])
+                version = koji_build["release"]
```

Comment:
This code block is almost identical to the code block above. Let's replace it with a method so we can adjust them together in the future if needed.